sklearn/utils/estimator_checks.py

```diff
@@ -252,6 +241,21 @@ def _check_transformer(name, Transformer, X, y):
             assert_raises(ValueError, transformer.transform, X.T)
 
 
+def check_fit_score_takes_y(name, Estimator):
+    # check that all estimators accept an optional y
+    # in fit and score so they can be used in pipelines
+    rnd = np.random.RandomState(0)
+    X = rnd.uniform(size=(10, 3))
+    y = (X[:, 0] * 4).astype(np.int)
+    y = multioutput_estimator_convert_y_2d(name, y)
+    with warnings.catch_warnings(record=True):
+        estimator = Estimator()
+    set_fast_parameters(estimator)
+    set_random_state(estimator)
+    estimator.fit(X, y)
```

Comment:
The ``sample_weights`` in ``DBSCAN`` mess with this. I'm pretty sure we need to change something. After your PR,
``make_pipeline(SelectKBest(), DBSCAN()).fit(X, y)`` will use ``y`` as sample weights in DBSCAN.